cpp/ql/src/Likely Bugs/Format/NonConstantFormat.ql

```diff
@@ -37,6 +37,37 @@ class UncalledFunction extends Function {
   }
 }
 
+/** The `unsigned short` type. */
+class UnsignedShort extends ShortType {
+  UnsignedShort() { this.isUnsigned() }
+}
+
+/**
+ * Holds if `t` cannot refer to a string. That is, it's a built-in
+ * or arithmetic type that is not a "`char` like" type.
+ */
+predicate cannotContainString(Type t) {
+  exists(Type unspecified |
+    unspecified = t.getUnspecifiedType() and
+    not unspecified instanceof UnknownType and
+    not unspecified instanceof CharType and
+    not unspecified instanceof WideCharType and
+    not unspecified instanceof Char8Type and
+    not unspecified instanceof Char16Type and
+    not unspecified instanceof Char32Type and
+    // C often defines `wchar_t` as `unsigned short`
+    not unspecified instanceof UnsignedShort
+  |
```

Comment:
In `nelson`, `wchar_t` appears to be a typedef for `int` (not even `unsigned int`).  I'm not sure if we want to try and expand this to all integral types though, or cut our losses here?

Reply:
Ouch that's strange. Thanks for spotting this. Let's just get this merged now and we can revisit this at some point later if necessary. I'll create an issue for it.

Reply:
I'm not really convinced it's worth fixing, unless it turns out to be common.

Reply:
Yeah, I agree